### ISSUE TITLE
db informations shown to visitor if bolt cannot connect to it or if rdbm crash

### ISSUE THREAD

**Issue by joanlouis** [bolt/bolt#3102]
i have already posted some times ago and it was corrected but the error come back :
```
PHP Fatal Error: Bolt Generic
Error: Uncaught exception 'Exception' with message 'exception 'PDOException' with message 'SQLSTATE[HY000] [2003] Can't connect to MySQL server on '127.0.0.1' (111)' in vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:43
Stack trace:
#0 vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php(43): PDO->__construct('mysql:host=127....', '**user**', '**password**', Array)
#1 vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php(45): Doctrine\DBAL\Driver\PDOConnection->__construct('mysql:host=127....', '**user**', '**password**', Array)
#2 vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(360): Doctrine\DBAL\Driver\PDOMySql\Driver->connect(Array, '**user**', '**password**', Array)
#3 src/Application.php(178): Doctrine\DBAL\Connection->connect()
#4 src/Application.php(159): Bolt\Application->checkDatabaseConnection()
#5 /home/web/it
File: src/Application.php
Line: 199
```

**Comment by GawainLynch** [comment]
Thank you!

**Comment by GawainLynch** [comment]
Problem found, fix incoming!

**Comment by joanlouis** [comment]
great !

**Comment by tobias2k** [comment]
You're still leaking information though - not the DB username and
password, but the fact that the problem lies in connecting to the
database, as well as which DBMS you're using. Neither is directly
exploitable, but can be beneficial for a number of attacks. Standard
practice is to serve a generic HTTP 500 error page and merely log that
there is a database connectivity problem.

Something could be said for displaying a more specific error message to
authenticated users, but because checking for authentication requires a
working database connection, this isn't possible here.

On Wed, Mar 11, 2015 at 04:42:05PM -0700, Gawain Lynch wrote:
> Closed #3102 via #3105.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/bolt/bolt/issues/3102#event-250544120

-- 
Tobias Dammers - tobias@twokings.nl - 070-3457628 - www.twokings.nl
Maandag t/m donderdag van 9.00 tot 17.30
Voor dringende vragen, mail naar support@twokings.nl

**Comment by GawainLynch** [comment]
> Standard practice is to serve a generic HTTP 500 error page and merely log that there is a database connectivity problem.

Sure, my first thought too… Problem is though, it's Chicken vs. Egg - :checkered_flag: The Ultimate Showdown! :checkered_flag:

The logging itself is to the database, so if the DB is down, no joy. The only place that we *should* have write access to is the cache and our [Silex debug logger](https://github.com/bolt/bolt/blob/master/app/config/config.yml.dist#L194) *could* be used… But how to best approach that is in itself a problem, as it can be very noisy and those log files (which are useless to most) just grow and grow.

In turn, just a 500 gives the person looking at the problem no idea what's wrong… Not saying we are doing this right, just can't see a "good" approach… your further thoughts?
